memory/player_party_manager.py

```diff
@@ -46,10 +52,50 @@ def update(self: Self) -> None:
                     self._read_gameobject_position()
                     self._read_movement_state()
                     self._read_leader_character()
+                    self._read_current_party()
+
             except Exception as _e:
                 # logger.debug(f"PlayerPartyManager Reloading {type(_e)}")
                 self.__init__()
 
+    def _read_current_party(self: Self) -> [PlayerPartyCharacter]:
+        try:
+            current_party_ptr = self.memory.follow_pointer(self.base, [0x98, 0x0])
+        except Exception:
+            self.players = []
+            return None
+
+        if current_party_ptr == self.NULL_POINTER:
+            self.current_party = []
+            return None
+        # Item is an array of pointers of size 0x08
+        items = self.memory.follow_pointer(
+            current_party_ptr,
+            [0x10, 0x0],
+        )
+        players = []
+
+        if items:
+            # Item objects are as follows:
+            # Items
+            #   - 0x18 - count
+            #   - 0x20 - Item[0]
+            #   - 0x28 - Item[1]
+            count = self.memory.read_int(items + 0x18)
+            address = self.ITEM_INDEX_0_ADDRESS
+
+            for _item in range(count):
+                ptr = self.memory.follow_pointer(items, [address, 0x0])
+                definition_id = self.memory.read_string(ptr + 0x14, 8)
+
+                # Definition IDS are stored as some goofy serialized utf encoded string
+                # We just do our best with the values that are provided to
+                # Determine the character we are looking at
+                players.append(PlayerPartyCharacter.parse_definition_id(definition_id))
+                address += self.ITEM_OBJECT_OFFSET
+            self.current_party = players
+        self.current_party = []
```

Comment:
I don't see an early return or else in this function, so doesn't this undo all the work we just did?